cpp/include/cudf/table/row_operators.cuh

```diff
@@ -29,6 +29,114 @@
 namespace cudf {
 namespace experimental {
 
+/**---------------------------------------------------------------------------*
+ * @brief Result type of the `element_relational_comparator` function object.
+ *
+ * Indicates how two elements `a` and `b` compare with one and another.
+ *
+ * Equivalence is defined as `not (a<b) and not (b<a)`. Elements that are
+ * EQUIVALENT may not necessarily be *equal*.
+ *
+ *---------------------------------------------------------------------------**/
+enum class weak_ordering {
+  LESS,        ///< Indicates `a` is less than (ordered before) `b`
+  EQUIVALENT,  ///< Indicates `a` is ordered neither before nor after `b`
+  GREATER      ///< Indicates `a` is greater than (ordered after) `b`
+};
+
+namespace detail {
+/**---------------------------------------------------------------------------*
+* @brief Compare the elements ordering with respect to `lhs`.
+*
+* @param[in] lhs first element
+* @param[in] rhs second element
+* @return weak_ordering Indicates the relationship between the elements in
+* the `lhs` and `rhs` columns.
+*---------------------------------------------------------------------------**/
+template <typename Element>
+__device__ weak_ordering compare_elements(Element lhs, Element rhs)
+{
+    if(lhs < rhs) {
+        return weak_ordering::LESS;
+    } else if(rhs < lhs) {
+        return weak_ordering::GREATER;
+    }
+    return weak_ordering::EQUIVALENT;
+}
+}
+/**---------------------------------------------------------------------------*
+* @brief A specialization for floating-point `Element` type rerlational comparison
+* to derive the order of the elements with respect to `lhs`. Specialization is to
+* handle `nan` in the order shown below.
+* `[-Inf, -ve, 0, -0, +ve, +Inf, NaN, NaN, null] (for null_order::AFTER)`
+* `[null, -Inf, -ve, 0, -0, +ve, +Inf, NaN, NaN] (for null_order::BEFORE)`
+*
+* @param[in] lhs first element
+* @param[in] rhs second element
+* @return weak_ordering Indicates the relationship between the elements in
+* the `lhs` and `rhs` columns.
+*---------------------------------------------------------------------------**/
+template <typename Element,
+            std::enable_if_t<std::is_floating_point<Element>::value>* = nullptr>
+__device__ weak_ordering relational_compare(Element lhs, Element rhs) {
+
+    if(std::isnan(lhs) and std::isnan(rhs)) {
```

Comment:
While I don't expect there to be any difference... I think we should use the `isnan` from the CUDA math lib as opposed to from `std`. 

https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1gf8093cd7c372f91c9837a82fd368c711